runtime/compiler/codegen/J9WatchedStaticFieldSnippet.cpp

```diff
@@ -88,7 +88,12 @@ uint8_t *TR::J9WatchedStaticFieldSnippet::emitSnippetBody()
             __LINE__,
             node);
          }
+#if defined(JITSERVER_SUPPORT)
+      // don't need to create relocation record for fieldClass on JITServer as it is already an client address
+      else if (!cg()->comp()->isOutOfProcessCompilation())
```

Comment:
On a second thought, I think `cg->needClassAndMethodPointerRelocations()` would be a better query here instead of `!cg()->comp()->isOutOfProcessCompilation()`. This will be consistent with other areas of the codegen as well.

For documentation, relocations for `TR_ClassAddress` are needed for AOT compiles as well as AOTaaS compiles. Meanwhile, they are not needed for regular JIT and JITServer compiles. `cg->needClassAndMethodPointerRelocations()` will give us the right answer on whether a relocation is needed depending on the type of compile being performed.